File: dmoj/usaco19opensilver1/main.cpp
#include <bitset>
#include <algorithm>
#include <iostream>

unsigned int n;
bool grid[1000][1000];
std::size_t rowCount[1000], columnCount[1000], count;

int main() {
    std::cin >> n;

    for (unsigned int r = 0; r < n; ++r) {
        std::string line;
        std::cin >> line;

        for (unsigned int c = 0; c < n; ++c)
            grid[r][c] = line[c] == 'R';
    }

    for (unsigned int i = 0; i < n; ++i) {
        if (grid[i][0])
            for (unsigned int j = 0; j < n; ++j)
                grid[i][j] = !grid[i][j];
        if (grid[0][i])
            for (unsigned int j = 0; j < n; ++j)
                grid[j][i] = !grid[j][i];
    }

    for (unsigned int i = 0; i < n; ++i) {
        for (unsigned int j = 0; j < n; ++j) {
            count += grid[i][j];
            rowCount[i] += grid[i][j];
            columnCount[i] += grid[j][i];
        }
    }

    if (count == (n - 1) * (n - 1))
        std::cout << 1 << ' ' << 1 << std::endl;
    else if (count == n - 1 && std::count(rowCount, rowCount + n, n - 1) == 1)
        std::cout << std::find(rowCount, rowCount + n, n - 1) - rowCount + 1 << ' ' << 1 << std::endl;
    else if (count == n - 1 && std::count(columnCount, columnCount + n, n - 1) == 1)
        std::cout << 1 << ' ' << std::find(columnCount, columnCount + n, n - 1) - columnCount + 1 << std::endl;
    else if (count == 1)
        std::cout << std::find(rowCount, rowCount + n, 1) - rowCount + 1 << ' ' << std::find(columnCount, columnCount + n, 1) - columnCount + 1 << std::endl;
    else
        std::cout << -1 << std::endl;

    return 0;
}

File: dmoj/usaco19opensilver1/main.py
from functools import partial
from operator import eq
from sys import stdin

n = int(stdin.readline())
grid = [list(map(partial(eq, 'R'), stdin.readline()[:-1])) for _ in range(n)]

for i in range(n):
    if grid[i][0]:
        for j in range(n):
            grid[i][j] = not grid[i][j]
    if grid[0][i]:
        for j in range(n):
            grid[j][i] = not grid[j][i]

count = sum(map(sum, grid))
row_count = tuple(map(sum, grid))
column_count = tuple(sum(grid[r][c] for r in range(n)) for c in range(n))

if count == (n - 1) ** 2:
    coordinates = 0, 0
elif count == n - 1 and row_count.count(n - 1) == 1:
    coordinates = row_count.index(n - 1), 0
elif count == n - 1 and column_count.count(n - 1) == 1:
    coordinates = 0, column_count.index(n - 1)
elif count == 1:
    coordinates = row_count.index(1), column_count.index(1)
else:
    coordinates = None

if coordinates is None:
    print(-1)
else:
    print(coordinates[0] + 1, coordinates[1] + 1)